experimental/dds/tree2/src/feature-libraries/editable-tree-2/proxies/proxies.ts

```diff
@@ -678,48 +716,145 @@ export function createRawObjectProxy<TSchema extends ObjectNodeSchema>(
  */
 export function extractFactoryContent<T extends ProxyNode<TreeNodeSchema, "javaScript">>(
 	content: T,
-): T {
-	if (Array.isArray(content)) {
-		// `content` is an array
-		return content.map(extractFactoryContent) as T;
+): ExtractedFactoryContent<T> {
+	if (isFluidHandle(content)) {
+		return { content, hydrateProxies: undefined };
+	} else if (Array.isArray(content)) {
+		return extractContentArray(content, 0);
```

Comment:
Nit: might be worth a quick comment explaining why this is always 0

Reply:
I documented the parameter and made it optional so it doesn't stick out here.